dicom_standard/extract_attributes.py

```diff
@@ -22,17 +25,17 @@ def get_attribute_table(standard):
     return attr_dict_list
 
 
-def attribute_table_to_list(table_div):
+def attribute_table_to_list(table_div: Tag) -> List[List[str]]:
     return [[cell.text.strip() for cell in row.find_all('td')]
             for row in pr.table_rows(table_div)]
 
 
-def attribute_table_to_json(table):
+def attribute_table_to_json(table: List[TableDictType]) -> List[TableDictType]:
     attributes = []
     for attr in table:
         attr['id'] = pl.create_slug(attr['tag'])
         attr['tag'] = attr['tag'].upper()
-        attr['retired'] = True if 'RET' in attr['retired'] else False
+        attr['retired'] = 'Y' if 'RET' in attr['retired'] else 'N'
```

Comment:
There was an error where `mypy` wouldn't allow me to add a boolean value to a `dict` of type `TableDictType` since they were supposed to be `Dict[str, str]`. If I used `Dict[str, Any]`, it wouldn't let me run `attr['tag'].upper()` since that's not a function for a boolean. I wasn't sure how to handle it so I tried a few things including attempting to make a TypedDict, and the only way I could get it to work was to copy the dictionary manually.

This also makes the format consistent with the `retired` property values in the confidentiality profile attributes table: http://dicom.nema.org/medical/dicom/current/output/html/part15.html#table_E.1-1